gradio/processing_utils.py

```diff
@@ -260,20 +270,51 @@ def save_file_to_cache(file_path: str | Path, cache_dir: str) -> str:
     return full_temp_file_path
 
 
+def check_public_url(url: str):
+    parsed_url = urlparse(url)
+    if parsed_url.scheme not in ["http", "https"]:
+        raise httpx.RequestError(f"Invalid URL: {url}")
+    hostname = parsed_url.hostname
+    if not hostname:
+        raise httpx.RequestError(f"Invalid URL: {url}")
+
+    try:
+        addrinfo = socket.getaddrinfo(hostname, None)
+    except socket.gaierror:
+        raise httpx.RequestError(f"Cannot resolve hostname: {hostname}") from None
```

Comment:
I was thinking about whether this would cause any issues if there's no internet connection, but I believe socket.getaddrinfo works for localhost addresses even without internet connections so this should be fine